api/rest/client/request.go

```diff
@@ -37,6 +38,44 @@ func (c *Client) doRequest(method, path string, body io.Reader) (*http.Response,
 	return c.client.Do(r)
 }
 
+// eventually we may want to trigger streaming with a boolean flag.
```

Comment:
are there any reasons that we'd want non-streaming file upload?

Reply:
My intention behind this comment was rather that we would have less code duplication if we only used `do` and `doRequest` functions (for file upload + all other requests) that take as an argument a streaming flag.

I am content leave the streaming specific functions for now though.

Reply:
ah ok i understand it now